components/malloc_size_of/lib.rs

```diff
@@ -825,6 +826,183 @@ impl<Static: string_cache::StaticAtomSet> MallocSizeOf for string_cache::Atom<St
     }
 }
 
+impl MallocSizeOf for usvg::Tree {
+    fn size_of(&self, ops: &mut MallocSizeOfOps) -> usize {
+        let root = self.root();
+        let linear_gradients = self.linear_gradients();
+        let radial_gradients = self.radial_gradients();
+        let patterns = self.patterns();
+        let clip_paths = self.clip_paths();
+        let masks = self.masks();
+        let filters = self.filters();
+        let fontdb = self.fontdb();
+
+        let mut sum = 0;
+        sum += root.size_of(ops);
+        sum += linear_gradients
+            .iter()
+            .map(|x| x.size_of(ops))
+            .sum::<usize>();
+        sum += radial_gradients
+            .iter()
+            .map(|x| x.size_of(ops))
+            .sum::<usize>();
+        sum += patterns.iter().map(|x| x.size_of(ops)).sum::<usize>();
+        sum += clip_paths.iter().map(|x| x.size_of(ops)).sum::<usize>();
+        sum += masks.iter().map(|x| x.size_of(ops)).sum::<usize>();
+        sum += filters.iter().map(|x| x.size_of(ops)).sum::<usize>();
+        sum += fontdb.size_of(ops);
```

Comment:
This is subtle, but `fontdb.size_of(ops)` is doing the wrong thing here, since `fontdb` is a `&Arc<FontDb>` (https://doc.servo.org/src/servo_malloc_size_of/lib.rs.html#175-180). Instead, we need `fontdb.conditional_size_of(ops)`.